internal/utils/retry.go

```diff
@@ -37,13 +55,15 @@ func (r *Retry) ExecWithRetries(ctx context.Context) error {
 
 			select {
 			case <-ctx.Done():
-				return context.Canceled
+				retryCtx.Errors = append(retryCtx.Errors, context.Canceled)
```

Comment:
@ctrombley should we update retryCtx.Canceled = true when context.Canceled?

Reply:
👍 I missed this, thanks.